exchanges/gateio/gateio_wrapper.go

```diff
@@ -641,21 +641,31 @@ func (e *Exchange) UpdateOrderbook(ctx context.Context, p currency.Pair, a asset
 
 // UpdateOrderbookWithLimit updates and returns the orderbook for a currency pair with a set orderbook size limit
 func (e *Exchange) UpdateOrderbookWithLimit(ctx context.Context, p currency.Pair, a asset.Item, limit uint64) (*orderbook.Book, error) {
+	book, err := e.fetchOrderbook(ctx, p, a, limit)
+	if err != nil {
+		return nil, err
+	}
+	if err := book.Process(); err != nil {
+		return nil, err
+	}
+	return orderbook.Get(e.Name, book.Pair, a)
+}
+
+func (e *Exchange) fetchOrderbook(ctx context.Context, p currency.Pair, a asset.Item, limit uint64) (*orderbook.Book, error) {
 	p, err := e.FormatExchangeCurrency(p, a)
 	if err != nil {
 		return nil, err
 	}
 	var o *Orderbook
 	switch a {
-	case asset.Spot, asset.Margin, asset.CrossMargin:
-		var available bool
-		available, err = e.checkInstrumentAvailabilityInSpot(p)
-		if err != nil {
+	case asset.Margin, asset.CrossMargin:
+		if available, err := e.checkInstrumentAvailabilityInSpot(p); err != nil {
```

Comment:
Why do we check the spot tradable pairs when working on margin and cross margin? could we not have a tradable pairs for these too even if the tradable pairs are similar? There may be cases when we need independent tradable pairs for margin and cross margin